### PR TITLE
Ido/fix auto fd mismatch

### PR DESCRIPTION
**Pull request by IdoWeka**

### CODE REVIEW

File: topics/MBC_cluster_conversion/mbc_divider_script.py [hunk 1]
```diff
@@ -131,6 +133,44 @@ def extract_digits(s):
     return "".join(filter(str.isdigit, s))
 
 
+def safer_drive_scan(drives_container_host_id, old_host_id):
+    retries = 15
+    drives_list = json.loads(run_shell_command('/bin/sh -c "weka cluster drive --host {} -J"'.format(old_host_id)))
+    drives_uuids = [disk["uuid"] for disk in drives_list]
+    print(drives_uuids)
```
Comment (danlweka) [inline topics/MBC_cluster_conversion/mbc_divider_script.py:140]: logger instead?

File: topics/MBC_cluster_conversion/mbc_divider_script.py [hunk 2]
```diff
@@ -131,6 +133,44 @@ def extract_digits(s):
...
+    scan_disk_command = '/bin/sh -c "weka cluster drive scan {}"'.format(drives_container_host_id)
+    run_shell_command(scan_disk_command)
+    sleep(2)
+    for i in range(retries):
+        drives_list = json.loads(run_shell_command('/bin/sh -c "weka cluster drive --host {} -J"'.format(old_host_id)))
+        drives_uuids = [disk["uuid"] for disk in drives_list]
+        if not drives_uuids:
+            return
+    logger.warning("the following drives did not move to the new container, {} retrying".format(drives_uuids))
+    raise
```
Comment (danlweka) [inline topics/MBC_cluster_conversion/mbc_divider_script.py:150]: `raise Exception("some error msg")` since there is no active exception.
or `quit(1)`

**Review by danlweka — CHANGES_REQUESTED** [review]

File: topics/MBC_cluster_conversion/mbc_divider_script.py [hunk 3]
```diff
@@ -131,6 +133,44 @@ def extract_digits(s):
...
+
+def check_and_fix_machine_identifier_in_failure_domain(host_id):
+    host_row = json.loads(run_shell_command('/bin/sh -c "weka cluster host {} -J"'.format(host_id)))[0]
+    failure_domain_machine_id = json.loads(run_shell_command(
+        '/bin/sh -c "weka debug config show failureDomains[{}].machineIdentifier"'
+        .format(extract_digits(host_row['failure_domain_id']))))
+    host_row_machine_id = json.loads(run_shell_command(
+        '/bin/sh -c "weka debug config show hosts[{}].machineIdentifier"'.format(host_id)))
+    if failure_domain_machine_id != host_row_machine_id:
+        logger.info('Fixing the machine identifier in {} to fit host {}'.format(host_row['failure_domain_id'], host_id))
```
Comment (danishpa) [inline topics/MBC_cluster_conversion/mbc_divider_script.py:161]: `logger.info` supports formatting syntax and passing as parameters of all parameters.

File: topics/MBC_cluster_conversion/mbc_divider_script.py [hunk 4]
```diff
@@ -276,6 +317,8 @@ def main():
     if old_failure_domain:
         failure_domain = '--failure-domain ' + old_failure_domain
         failure_domain_for_local = '--name ' + old_failure_domain
+    else:
+        check_and_fix_machine_identifier_in_failure_domain(current_host_id)
     for slot in prev_resources['nodes']:
```
Comment (danishpa) [inline topics/MBC_cluster_conversion/mbc_divider_script.py:322]: Newline here

File: topics/MBC_cluster_conversion/mbc_divider_script.py [hunk 5]
```diff
@@ -14,10 +14,12 @@
 from concurrent.futures import ThreadPoolExecutor
 from resources_generator import GiB
 
-logging.basicConfig(format='%(asctime)s %(levelname)s: %(message)s',
-                    datefmt='%m/%d/%Y %I:%M:%S %p')
 logger = logging.getLogger('mbc divider')
-
+ch = logging.StreamHandler()
```
Comment (danishpa) [inline topics/MBC_cluster_conversion/mbc_divider_script.py:18]: You don't need to specify stream?

File: topics/MBC_cluster_conversion/mbc_divider_script.py [hunk 6]
```diff
@@ -131,6 +133,44 @@ def extract_digits(s):
...
+        kv_strings_list = []
+        for i in range(len(host_row_machine_id)):
+            kv_strings_list.append("machineIdentifier[{}]={}".format(i, host_row_machine_id[i]))
+        config_assign_cmd = '/bin/sh -c "weka debug config assign failureDomains[{}] {}"'\
+            .format(extract_digits(host_row["failure_domain_id"]), " ".join(kv_strings_list))
+        try:
+            run_shell_command(config_assign_cmd)
+        except Exception as e:
+            logger.warning("Failed to change the machine identifier for {} with the the following error: {}".format(host_row["failure_domain_id"], e))
+            raise
```
[the remark this answers is not in the document]
Comment (IdoWeka) [reply topics/MBC_cluster_conversion/mbc_divider_script.py:171]: noted

File: topics/MBC_cluster_conversion/mbc_divider_script.py [hunk 7]
```diff
@@ -131,6 +133,44 @@ def extract_digits(s):
...
+    for i in range(retries):
+        drives_list = json.loads(run_shell_command('/bin/sh -c "weka cluster drive --host {} -J"'.format(old_host_id)))
+        drives_uuids = [disk["uuid"] for disk in drives_list]
+        if not drives_uuids:
+            return
+    logger.warning("the following drives did not move to the new container, {} retrying".format(drives_uuids))
+    raise
+
+
+def check_and_fix_machine_identifier_in_failure_domain(host_id):
```
[the remark this answers is not in the document]
Comment (IdoWeka) [reply topics/MBC_cluster_conversion/mbc_divider_script.py:153]: there is a way to fix the product... but i need to discuss it with you..

File: topics/MBC_cluster_conversion/mbc_divider_script.py (hunk 5 above)
Comment (IdoWeka) [reply topics/MBC_cluster_conversion/mbc_divider_script.py:18]: the auto stream is stdout

**Review by danlweka — APPROVED** [review]

**Pull request merged**